controllers/iampolicy_controller.go

```diff
@@ -464,30 +465,34 @@ func addViolationCount(plc *iampolicyv1.IamPolicy, roleName string, userCount in
 	return true
 }
 
-func checkComplianceBasedOnDetails(plc *iampolicyv1.IamPolicy, roleName string) {
+func checkComplianceBasedOnDetails(plc *iampolicyv1.IamPolicy, roleName string) bool {
```

Comment:
Could you please document the function since the return value isn't clear unless you look at how it's called?